pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/BaseFilterOperator.java

```diff
@@ -56,4 +73,8 @@ public final BaseFilterBlock getNextBlock(BlockId blockId) {
   }
 
   public abstract BaseFilterBlock nextFilterBlock(BlockId blockId);
+
+  public boolean alwaysFalse() {
```

Comment:
this should be abstract and let the actual filter operator implement it appropriately. The way this works for sorted , inverted, scan can be completely different. It should not be limited to using predicatEvaluator. We can potentially use bloomfilter for no dictionary columns to compute isAlwaysFalse.